tests/validation/cattlevalidationtest/core/test_native_docker.py

```diff
@@ -59,164 +59,193 @@ def pull_images(docker_client):
+        assert container.state == 'running'
+
+        container = admin_client.wait_success(container.stop(timeout=0))
+        assert container.state == 'stopped'
+
+        container = admin_client.wait_success(container.remove(timeout=0))
+        assert container.state == 'removed'
+
+        container = admin_client.wait_success(container.purge(timeout=0))
+        assert container.state == 'purged'
+    finally:
+        cleanup_container(admin_client, container)
 
 
 def test_native_not_started(docker_client, admin_client, super_client,
                             pull_images):
-    name = 'native-%s' % random_str()
-    d_container = docker_client.create_container(TEST_IMAGE,
-                                                 name=name,
-                                                 environment=[
-                                                     'RANCHER_NETWORK=true'])
+    try:
+        name = 'native-%s' % random_str()
+        d_container = docker_client. \
+            create_container(TEST_IMAGE, name=name,
+                             environment=['RANCHER_NETWORK=true'])
 
-    def check():
-        containers = admin_client.list_container(name=name)
-        return len(containers) > 0
+        def check():
+            containers = admin_client.list_container(name=name)
+            return len(containers) > 0
 
-    wait_for(check, timeout_message=CONTAINER_APPEAR_TIMEOUT_MSG % name)
+        wait_for(check, timeout_message=CONTAINER_APPEAR_TIMEOUT_MSG % name)
 
-    r_containers = admin_client.list_container(name=name)
-    assert len(r_containers) == 1
-    c = r_containers[0]
-    c = admin_client.wait_success(c)
-    c_id = c.id
+        r_containers = admin_client.list_container(name=name)
+        assert len(r_containers) == 1
+        container = r_containers[0]
+        container = admin_client.wait_success(container)
+        c_id = container.id
 
-    assert c.externalId == d_container['Id']
-    assert c.state == 'running'
+        assert container.externalId == d_container['Id']
+        assert container.state == 'running'
 
-    def stopped_check():
-        c = admin_client.by_id_container(c_id)
-        return c.state == 'stopped'
+        def stopped_check():
+            c = admin_client.by_id_container(c_id)
+            return c.state == 'stopped'
 
-    wait_for(stopped_check,
-             'Timeout waiting for container to stop. Id: [%s]' % c_id)
+        wait_for(stopped_check,
+                 'Timeout waiting for container to stop. Id: [%s]' % c_id)
 
-    nics = super_client.reload(c).nics()
-    assert len(nics) == 1
-    assert c.primaryIpAddress == nics.data[0].ipAddresses().data[0].address
+        nics = super_client.reload(container).nics()
+        assert len(nics) == 1
+        assert container.primaryIpAddress == nics.data[0].ipAddresses().data[
+            0].address
+    finally:
+        cleanup_container(admin_client, container)
```

Comment:
quick question: in the other 2 tests, we declare the container to be None outside of the try block.  i'm still a python newbie.  do we need to do that here too?